Bottom: b0ecaac03a4cee3fc35723b2693d5f56a0be1c5b
Top:    8cc8c3b36a778681aaf5fe8295deb292cce4df06
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-12-22 20:35:09 +0000

hid/gtk: Disable GL alpha-blending for the pinout preview

We aren't using subcompositing for this, so don't alpha-blend.


---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index dad2d61..0467d0e 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -807,6 +807,8 @@ ghid_init_drawing_widget (GtkWidget *widget, GHidPort *port)
 {
   render_priv *priv = port->render_priv;
 
+  GTK_WIDGET_UNSET_FLAGS (GTK_WIDGET (widget), GTK_DOUBLE_BUFFERED);
+
   gtk_widget_set_gl_capability (widget,
                                 priv->glconfig,
                                 NULL,
@@ -1083,15 +1085,23 @@ ghid_pinout_preview_expose (GtkWidget *widget,
   ghid_start_drawing (gport, widget);
   hidgl_start_render ();
 
-  glEnable (GL_BLEND);
-  glBlendFunc (GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
+//  glEnable (GL_BLEND);
+//  glBlendFunc (GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
 
   glViewport (0, 0, allocation.width, allocation.height);
 
+#if 0
   glEnable (GL_SCISSOR_TEST);
   glScissor (ev->area.x,
              allocation.height - ev->area.height - ev->area.y,
              ev->area.width, ev->area.height);
+  printf ("EVT: %i, %i, w=%i, h=%i, Scissor setup: glScissor (%f, %f, %f, %f);\n",
+          ev->area.x, ev->area.y, ev->area.width, ev->area.height,
+             (double)ev->area.x,
+             (double)(allocation.height - ev->area.height - ev->area.y),
+             (double)ev->area.width,
+             (double)ev->area.height);
+#endif
 
   glMatrixMode (GL_PROJECTION);
   glLoadIdentity ();
